samtranslator/intrinsics/actions.py

```diff
@@ -503,3 +503,50 @@ def _get_resolved_dictionary(self, input_dict, key, resolved_value, remaining):
             input_dict[key] = [resolved_value] + remaining
 
         return input_dict
+
+
+class FindInMapAction(Action):
+    """
+    This action can't be used along with other actions.
+    """
+    intrinsic_name = "Fn::FindInMap"
+
+    def resolve_parameter_refs(self, input_dict, parameters):
+        """
+        Recursively resolves "Fn::FindInMap"references that are present in the mappings and returns the value.
+        If it is not in mappings, this method simply returns the input unchanged.
+
+        :param input_dict: Dictionary representing the FindInMap function. Must contain only one key and it
+                           should be "Fn::FindInMap".
+
+        :param parameters: Dictionary of mappings from the SAM template
+        """
+        if not self.can_handle(input_dict):
+            return input_dict
+
+        value = input_dict[self.intrinsic_name]
+
+        if not isinstance(value, list) or len(value) != 3:
```

Comment:
Can you add a comment, like "FindInMap expects an array with 3 values". Wondering if we should add some validation here and throw?

Reply:
Do you mean we should throw `InvalidTemplateException` here instead of silently return the input?

Reply:
Right. I'm wondering in what circumstances would the user want it to silently return instead of having some validation which will warn them the template is invalid before it creates/executes the change set.